chat/views/get_message_count_by_date.py

```diff
@@ -83,51 +92,51 @@ def get_message_count_by_bot(request):
     if not email:
         return generate_error_response(400, "Authentication is required")
 
-    # Get the bot's ID from the request parameters
+    # Get the parameters
     bot_id = request.GET.get('botId', None)
-    if bot_id is None:
-        return generate_error_response(400, "Bot ID is required")
-
-    # Check if the bot ID is a valid UUID
-    try:
-        uuid.UUID(bot_id)
-    except ValueError:
-        return generate_error_response(400, "Invalid Bot ID")
+    sorting_order = request.GET.get('sort', DESCENDING_ORDER)
+    num_days = request.GET.get('daysLimit', None)
 
-    # Get sorting order from URL parameter
-    sorting_order = request.GET.get('sort', 'desc')
+    # Parameter checks
+    if bot_id is not None:
+        try:
+            # Check if the bot ID is a valid UUID
+            uuid_obj = uuid.UUID(bot_id)
+        except ValueError:
+            return generate_error_response(400, "Invalid Bot ID")
+        # Check if the bot exists
+        if not ChatBot.objects.filter(id=uuid_obj).exists():
+            return generate_error_response(404, "Bot not found")
 
-    # Get the value of num_days from the query parameters
-    num_days = request.GET.get('daysLimit', None)
     if num_days is not None:
         try:
             num_days = int(num_days)
+            start_date = date.today() - timedelta(days=num_days)
         except ValueError:
             return generate_error_response(400, "Invalid value for 'num_days'. Must be an integer.")
```

Comment:
num_days is an internal variable, I think the error message should reference daysLimit here.